src/scripts/bash/k8s/index-delta.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+# Get the delta of objects created or updated since a certain time, and index them
+# See Issue #1990: https://github.com/NCEAS/metacat/issues/1990
+#
+# Use case: moving a Metacat instance to a Kubernetes cluster, we rsync the data initially to
+# cephfs (at time 't'), then do all the setup and indexing while the legacy metacat is still
+# online. Then, some time later (t + dt), we make the legacy host read-only, and do a "top-up" rsync
+# again to cephfs. We then need to index only the objects that have been created or updated between
+# t and t + dt.
+#
+
+cleanup() {
+    rm -f "${cert_file}"
+    rm -f "${result_file}"
+
+}
+
+# Initialize vars
+
+if [ $# -ne 2 ]; then
+    echo "Usage: $0 <start-time>"
+    echo "       where <start-time> is the time an hour or more before the previous rsync,"
+    echo "              in the format: yyyy-mm-dd HH:MM:SS (with a space; e.g. 2024-11-01 14:01:00)"
+    exit 1
+fi
+
+start_time="$1 $2"
+release_name=$(helm ls | grep "metacat" | awk  '{print $1}')
+rls_count=$(echo ${release_name} | wc -w)
+
+if [ "${rls_count}" -ne "1" ]; then
+    echo "Can't get release name; found ${rls_count} releases containing 'metacat': ${release_name}"
+    echo "Please type the correct release name (or Ctrl+C to abort):"
+    echo
+    read release_name
+fi
+
+cert_file="_DELETEME_$(date +%Y-%m-%d)_node.crt"
+kubectl get secret ${release_name}-d1-client-cert -o jsonpath="{.data.d1client\.crt}" | base64 -d \
+    > $cert_file
+chmod 600 $cert_file
+
+host=$(kubectl get ingress | grep "${release_name}" | awk '{print $3}')
+api_url="https://${host}/metacat/d1/mn/v2/index"
+
+result_file="modified-since-$(echo -n ${start_time} | sed -e 's/ /-/g').txt"
+
+
+# Get the pids of the objects modified since the start time
+echo
+echo "Finding and indexing any objects created/modified since ${start_time}, in release ${release_name}."
+echo
+echo "Double-check that ${start_time} is the time an hour or more before the previous rsync, and is in"
+echo "the yyyy-mm-dd HH:MM:SS format (e.g. 2024-11-01 14:01:00)"
+echo
+echo "Results will be saved to ${result_file}. Enter to continue or Ctrl+C to abort..."
+read
+
+kubectl exec ${release_name}-postgresql-0 -- bash -c "psql -U metacat << EOF
+  SELECT guid FROM systemmetadata WHERE date_modified > '${start_time}';
+EOF" | sed '1,2d;$d' | sed '$d'  >  ${result_file}
+# sed removes first 2 rows: "guid" and "----", and the last 2 rows: "(n rows)" and <blank>
+
+cat ${result_file}
+echo
+echo "PIDs of modified objects saved to ./${result_file}"
+echo
+
+
+# Now, index the objects
+result_count=$(cat ${result_file} | wc -w | xargs)
+echo "Found ${result_count} objects created/modified since ${start_time}, in release"
+echo "${release_name}."
+echo
+echo "Calling ${api_url} to index each. Enter to continue or Ctrl+C to abort..."
+read
+
+idx=0
+for pid in $(cat ${result_file}); do
```

Comment:
Is it possible to put two or thee pids into one curl command so we can cut some network traffics? It depends on how many pids need to be indexed. If it is just dozens, maybe it is not worth to do it.

Reply:
If there are a lot of PIDs, `parallel` or `xargs -j` would likely be much faster than the serial loop. See: https://learning.nceas.ucsb.edu/2024-03-arctic/sections/parallel-programming.html#parallel-processing-in-the-shell

Reply:
Thanks. The requests are very fast (they just queue the pids for indexing then return), and I don't foresee there being enough pids to warrant parallelizing (eg for ADC, there were just over 3k pids to reindex in the last month).

Reply:
batched in groups of 5 pids per call